### PR TITLE
(#14) add dev environment

### PR DESCRIPTION
**Pull request by AndreasAugustin**
Add developer docker environment with [katacoda-cli](https://www.katacoda.com/cli).

### CODE REVIEW

**Comment by AndreasAugustin** [comment]
Could be also used for `ci` tasks. to prevent  
For with travis (.travis.yaml):

```yaml
sudo: required
services:
  - docker

env:
  COMPOSE_VERSION: 1.17.1

before_install:
 - curl -L https://github.com/docker/compose/releases/download/${COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
 - chmod +x docker-compose
 - sudo mv docker-compose /usr/local/bin
 - docker-compose version

script:
  - make docker-build
  - make validate
```

**Comment by AndreasAugustin** [comment]
Addition to #15

File: .dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM node:12.6-alpine
+
+RUN apk add --update --no-cache bash make
+RUN npm install npm install katacoda-cli -g
```
Comment (BenHall) [inline .dev/Dockerfile]: Could we clean up the double npm install please?

**Comment by BenHall** [comment]
This is cool. I like it!! The validate feature was designed to be included in a CI system so nice to have an example.

If you could update the Dockerfile we'll merge this in :)

File: .dev/Dockerfile (hunk 1 above)
Comment (AndreasAugustin) [reply .dev/Dockerfile]: Oh thx!
First I included there also [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) but decided that this should be not part of this PR.
Did not clean up all of the stuff :man_facepalming: (should not hurry.. :) )

- [ ] cleanup

**Pull request merged**